rfq/negotiator.go

```diff
@@ -405,8 +405,9 @@ func (n *Negotiator) HandleIncomingSellRequest(
 		// Query the price oracle for a bid price. This is the price we
 		// are willing to pay for the asset that our peer is trying to
 		// sell to us.
+		assetID, assetGroupKey := request.AssetSpecifier.UnwrapToPtr()
 		assetRate, rateExpiry, err := n.queryBidFromPriceOracle(
-			request.Peer, request.AssetID, request.AssetGroupKey,
+			request.Peer, assetID, assetGroupKey,
```

Comment:
Should we instead pass in the `Specifier` into `queryBidFromPriceOracle` to avoid some duplication? Then we likely also don't have to unwrap to pointer?

Reply:
`queryBidFromPriceOracle` is called given a `BuyRequest` in another part of the code. And the `BuyRequest` does not have an `AssetSpecifier` field yet. I will add and then `queryBidFromPriceOracle` can accept an `asset.Specifier`.

Reply:
I notice now that `BuyOrder` will need the asset specifier upgrade at the same time as `queryBidFromPriceOracle`. I'm just going to leave that for a future PR. It distracts from the main goal of this PR, IMO.